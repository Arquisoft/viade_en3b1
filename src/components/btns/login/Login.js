import React from 'react';
import { Button } from '@material-ui/core';
import { LoggedIn, LoggedOut } from '@solid/react';

const auth = require('solid-auth-client');

export class Login extends React.Component {

    async popupLogin(auth) {
        let session = await auth.currentSession();
        let popupUri = 'https://solid.community/common/popup.html';
        if (!session) {
            session = await auth.popupLogin({ popupUri });
        }

        // if(session){
        //     UsersManager.createUserAndLoadRoutes(`${session.webId}`);
        // }
        return (`${session.webId}`); // example of getting user's name. returns something like: "https://pablocanalsuarez.solid.community/profile/card#me"
    };

    logout(auth) {
        auth.logout();
    }

    render() {
        return (
            <div>
                <LoggedOut>
                    <Button variant="outlined" color="inherit" onClick={() => this.popupLogin(auth)}>Sign in</Button>
                </LoggedOut>
                <LoggedIn>
                    <Button variant="outlined" color="inherit" onClick={() => this.logout(auth)}>Sign out</Button>
                </LoggedIn>
            </div>
        );
    }
}

export default Login;