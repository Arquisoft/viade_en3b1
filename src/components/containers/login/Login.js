import React from 'react';
import {  Container, Card, CssBaseline, makeStyles, CardContent } from '@material-ui/core';
import { ProviderLogin } from "@inrupt/solid-react-components";
import NavBar from '../../ui/main/NavBar';

const useStyles = makeStyles((theme) => ({
    root: {
        marginTop: theme.spacing(20)
    }
}));

export default function Login() {

    const classes = useStyles();

    return (
        <div>
        <NavBar />
            <Container component="main" maxWidth="sm">
                <Card className={classes.root} elevation={4}>
                    <CssBaseline />
                    <CardContent>
                        <ProviderLogin
                            className='solid-provider-login-component'
                            callbackUri={`${window.location.protocol}//${window.location.host}/viade_en3b1/#/home`}
                            selectPlaceHolder='Select your Provider'
                            inputPlaceholder='Introduza la url de su webId'
                            formButtonText='Login'
                            btnTxtWebId='Log In with WebId'
                            btnTxtProvider='Log In with Provider'
                            errorsText={{
                                unknown: 'Something is wrong, please try again...',
                                webIdNotValid: 'WebID is not valid',
                                emptyProvider: 'Solid Provider is required',
                                emptyWebId: 'Valid WebID is required'
                            }}
                        />
                        </CardContent>
                </Card>
            </Container>
        </div>
    );

}