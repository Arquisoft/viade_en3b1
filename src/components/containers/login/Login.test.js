import Login from './Login';

test('Define Login', () => {
  expect(Login).toBeDefined();
});

test('Renders correctly', () => {
  const wrapper = render(<Login />);
  expect(wrapper).toMatchSnapshot();
});