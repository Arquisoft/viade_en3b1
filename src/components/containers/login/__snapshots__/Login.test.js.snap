// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <div
    class="makeStyles-root-2"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-3 MuiToolbar-gutters"
      >
        <div
          class="MuiGrid-root makeStyles-link-5"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
              href="#/home"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <img
                    alt="Viade Logo"
                    src="logo_2.svg"
                    style="height:2.5rem;width:2.5rem"
                  />
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4"
                  >
                    ViaDe
                  </h4>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation4 MuiPaper-rounded"
      style="min-height:15rem"
    >
      <div
        class="MuiCardContent-root"
      >
        <div
          class="sc-Axmtr hNtgwq solid-provider-login-component solid-provider-login-component null"
        >
          <form>
            <input
              class="sc-AxhCb deVEyJ"
              data-testid="input-webid"
              name="idp"
              placeholder="Introduza la url de su webId"
              type="text"
            />
            <button
              class="sc-AxhUy cLoezo link "
              data-testid="change-mode-button"
              type="button"
            >
              Log In with Provider
            </button>
            <button
              class="sc-AxgMl gigcwx"
              data-testid="provider-form-button"
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </main>
</div>
`;
